beebop/app.py

```diff
@@ -147,51 +154,74 @@ def add_files(
 
 
 @app.errorhandler(500)
-def internal_server_error(e) -> json:
+def internal_server_error(e) -> tuple[Response, Literal[500]]:
     """
     :param e: [error]
-    :return json: [error response object]
+    :return Response: [error response object]
     """
+    logger.warning(f"Internal Server Error: {e}")
     return (
         jsonify(
             error=response_failure(
-                {"error": "Internal Server Error", "detail": str(e)}
+                ResponseError(
+                    error="Internal Server Error", detail=str(e.description)
+                )
             )
         ),
         500,
     )
 
 
+@app.errorhandler(400)
+def bad_request(e) -> tuple[Response, Literal[400]]:
+    """
+    :param e: [error]
+    :return Response: [error response object]
+    """
+    logger.warning(f"Bad Request: {e}")
+    return (
+        jsonify(
```

Comment:
I'm a little bit confused by this use of jsonify - I thought you set the status through the jsonify method itself, but here status is going out separately in the tuple. And not sure why the response_failure has to be set as the value of "error" here when you've already defined the response body format in `response_failure`. What effect does that have?